---
layout: page
title: Projects
permalink: /Projects/
---

Our lab focuses on understanding the genetic basis of neurodegenerative
disease (i.e., Alzheimer Disease) and psychological traits (i.e., psychological
well-being or resilience). Our work has been featured in the
    <a href="http://www.huffingtonpost.com/entry/593de662e4b0b65670e56be6">
      Huffington Post</a>, the
    <a href="http://emoryhealthmagazine.emory.edu/issues/2016/winter/features/blessing-from-a-curse/index.html">
      Emory Health Magazine</a>, and
    <a href="https://www.cnn.com/2019/02/06/health/education-dementia-study/index.html">CNN
    </a>.

### Understanding the molecular mechanisms behind how psychological well-being and depression modify AD risk

Psychological well-being (PWB) and depression are important factors that modify
risk for Alzheimer's disease (AD). Specifically, depression is associated with
increased risk for AD while PWB with decreased risk for AD. Molecular mechanisms
underlying these important associations, however, are not known. To address this
knowledge gap, we will study a unique dataset of 850 human postmortem brains
from the Rush Memory and Aging Project. This prospective longitudinal project
annually collects data on depression, PWB, cognition, physical health, and
dementia, and genomic, transcriptomic, and proteomic data from the dorsolateral prefrontal cortex (dPFC). We will identify key miRNAs, transcripts, and proteins associated with depression, and separately with PWB, and examine how they relate
to cognitive change, AD dementia, and dementia-related pathologies.

### Identifying new Genetic causes of Alzheimer's Disease

- **Identifying New Genetic causes of Early-onset Alzheimer's disease** -
Early-onset Alzheimer's disease is a particularly aggressive form of the disease with higher heritability than typical late-onset AD and has several well-described
genetic causes. We use next-generation sequencing to identify new genetic causes
in families and individuals with this particularly aggressive form of AD.

- **A genetic study of a large pedigree with late-onset Alzheimer's Disease** -
late-onset Alzheimer's Disease, the more common form of the disease, is due to
the cumulative effect of 1,000s or 10,000s of genetic changes. However, there
still appear to be families that transmit the illness as an autosomal dominant
trait. The Emory ADRC has collected a number of very large families and using
linkage and next-generation sequencing techniques we are moving to identify
high-risk alleles in these families.

### A Proteogenomic Approach to Understanding Alzheimer's Disease Genome-wide Association Study Results

Recent genome wide association studies (GWAS) of AD have identified at least >20
regions of the human genome that are linked with developing AD. Currently, it is
not known why these regions are associated with AD. Our overarching hypothesis
is that some AD GWAS signals are due to one or more coding variants that makes
the translated protein more likely to aggregate. We are uniquely poised to
address this hypothesis given the unique datasets of brain tissue and our
expertise in both large-scale and targeted proteomic analyses. By combining our
genetic sequencing data with cutting-edge mass spectrometry we will sequence
encoded protein products of the same genes from individuals that underwent
genetic sequencing. This will allow us to ask which genetic variants associate
with AD and whether those genetic variants influence the abundance or
aggregation potential of proteins in the brains of individuals with AD.
Furthermore, we have developed a new method to detect and measure novel proteins
that result from genetic sequence variants that cause a change the primary amino
acid sequence. This gives us the ability to directly test whether variant
containing proteins are more or less abundant or aggregation prone in carriers
versus non-carriers and determine whether that contributes to AD risk.

### Accelerated aging and risk for Alzheimer’s disease in the Emory Healthy Brain and Emory Healthy Aging Studies

Accelerated aging occurs when a person’s biological age (as estimated by epigenetic DNA modification) is greater than their chronological age. In this project, we will investigate if accelerated aging is associated with lower cognitive performance, higher rate of biomarker evidence for neurodegenerative disease, and higher genetic risk for Alzheimer’s disease. In addition, we will investigate whether accelerated aging is associated with mood (anxiety, depression), resilience, purpose-in-life, and local air quality.

### Elucidating molecular mechanisms of psychological well-being

Psychological well-being (PWB) is a multidimensional construct that encompasses positive emotion, life satisfaction, 
and sense of purpose and meaning in life, and is more than the absence of negative emotional states. 
Many prospective longitudinal studies have shown that PWB is associated with better mental and physical health 
after adjusting for negative emotion and other relevant confounding factors. Indeed, PWB mitigates risks 
of having suicide ideation, depression, substance abuse, post-traumatic stress disorder, Alzheimer’s dementia, 
heart disease, diabetes, and reduces all-cause mortality. In this project we will investigate molecular mechanisms
that underlie individual differences in PWB using human post-mortem brain multi-omic data.

### Transgenerational effects of maternal stressors: Investigating the role of infant gene expression

Several prospective, longitudinal, mother-child cohort studies have found that children exposed to maternal
psychological stress, depression, or anxiety during the prenatal period have higher risk for behavioral 
and emotional problems later in life, including increased fearfulness, anxiety, and depression. 
We will investigate molecular mechanisms underlying this association in collaboration with investigators 
at University of Cape Town in South Africa and at McLean Hospital.

### A brain multi-omic approach to identify key molecular drivers of neuropsychiatric symptoms in Alzheimer's dementia (R01 AG072120) 

Approximately 65% of individuals with mild cognitive impairment (MCI), Alzheimer's disease (AD), or AD related dementias (ADRD) experience neuropsychiatric symptoms (NPS). These debilitating symptoms include depression, anxiety, apathy, delusions, hallucinations, agitation, sleep disturbances and are associated with faster disease progression, greater functional impairment, higher caregiver burden, and earlier institutionalization. Current treatments for NPS in MCI/dementia have limited efficacy but high rates of adverse side effects, including higher mortality. Therefore, safe and effective treatments for NPS are urgently needed. However, we have limited insights into molecular mechanisms of NPS in MCI/dementia to nominate therapeutic targets. To address this knowledge gap, we aim to elucidate the genetic and molecular mechanisms underlying NPS in MCI/dementia using two complementary but independent approaches. 

In the first approach, we will leverage the substantial genetic basis of NPS (heritability of ~61%) to identify novel genes involved in NPS. First, we will perform genome-wide association studies (GWAS) of NPS in ~32,000 individuals with MCI/AD/ADRD from across the United States. We will perform a GWAS of each NPS domain as well as a multivariate GWAS of all NPS collectively. Results of each GWAS will be integrated with reference human brain transcriptomic and proteomic profiles to identify individual brain transcripts and proteins that are consistent with being causal in NPS. We have applied these integrative approaches to AD and identified 11 genes that regulate their brain protein abundances to predispose to AD (Wingo et al. 2021, Nat Genet). Unlike GWAS results, these findings point to specific brain proteins that are consistent with a causal role in AD. Importantly, this approach does not solely depend on genome-wide significant signals, and the causal inference of this integrative strategy has been experimentally tested and found to be robust. Thus, we expect that findings from this approach will provide novel insights into the pathogenesis of NPS. 

Our second approach aims to identify brain transcripts, proteins, and co-expression networks associated with NPS. We will perform transcriptomic and proteomic sequencing of post-mortem brain tissues from 400 individuals with MCI/AD and longitudinal NPS assessment, as existing brain omics data do not have robust NPS measures. Then we will examine NPS domains individually as well as collectively in our differential expression and network analyses. We anticipate that molecular alterations at the brain transcript and protein levels identified with this approach will provide insights into NPS pathogenesis and progression. Studying both transcripts and proteins would provide complementary information and a more complete molecular picture. We view understanding proteins as particularly important since they are the vast majority of drug targets.  

The proposed project will likely lead to new insights into molecular mechanisms and novel therapeutic targets for NPS and thereby have an important and sustained impact on public health. 

### Integrative genomic, transcriptomic, and proteomic analyses to investigate sex-specific differences in Alzheimer's disease (R01 AG075827) 

Alzheimer's disease (AD) affects 35 million people worldwide. However, there is no effective treatments to slow or halt the underlying neurodegeneration of AD. Strikingly, women are affected by AD about twice as much as men. Why women are disproportionately affected by AD is not well understood. 

Here, we hypothesize that there is an interaction between biological sex and brain gene expression that predisposes women to have a higher risk for AD. This hypothesis builds on our work investigating the genetic control of gene expression in the human brain. In those studies, we have identified genetic sites that are associated with variation in gene expression at the transcript and protein level in the human brain. Identifying genetic sites associated with natural variation in brain gene expression are incredibly useful to resolve genetic signals identified by genome-wide association study (GWAS) to specific causal transcripts and proteins. We have successfully used brain gene transcript and protein expression results with GWAS results to identify novel proteins for AD (Wingo et al, Nat Genet, 2021) and depression (Wingo et al, Nat Neurosci, 2021). These analyses implicate the causes of inherited genetic risk, which are presumably among the most important early contributors to disease, which make them highly relevant to resolving sex-specific disease risk. 

To test our hypothesis, we will first perform a sex-specific brain expression analysis for transcripts and proteins (Aim 1). To do this, we will leverage existing brain transcripts and protein expression data generated by national resources, and we propose a novel approach to cost-effectively increase the depth of proteomes to improve power. We expect to generate sex-specific quantitative trait loci (QTL) for brain transcripts (N=1655, 61% women) and proteins (N=1584, 66% women), and each gene will then be tested for evidence of interaction with sex. These results are likely to be of general interest to the neuroscience and genetics communities, and the data and results will be made widely available to the science community. In Aim 2, we will perform a sex-specific GWAS of AD, and we will identify genetic variants that interact with sex. To do this, we will perform sex-stratified GWAS using 1) case-control status for AD in participants from US studies (N=27,580), 2) AD-proxy case-control in participants from the UK Biobank (N=431,000), and iii) a meta-analysis of 1 and 2. In Aim 3, we will identify brain transcripts and proteins contributing to AD pathogenesis that are specific to men or women, respectively. To do this, we will integrate the sex-specific brain QTLs with the sex-specific GWAS results to resolve GWAS signals to proteins for women and men, separately. For AD genetic signals with evidence for sex interaction, we will test whether sex-specific gene expression accounts for the differences in AD risk between the sexes using causal inference approaches (e.g., Mendelian randomization). Findings from this project are highly likely to provide novel mechanistic insights into sex differences in AD and promising new targets for further sex-specific mechanistic and therapeutic studies of AD. 

### Identifying novel brain proteins contributing to PTSD and alcohol use disorder (VA, I01 BX005686) 

PTSD and alcohol use disorder (AUD) are among the most prevalent and debilitating psychiatric disorders in Veterans. They are highly comorbid and have shared genetic susceptibility. Treatments for PTSD or AUD are ineffective in many patients, and comorbid PTSD and AUD are often more difficult to treat and associated with more severe symptoms, higher suicide risk, and poorer outcomes. Thus, the overarching goal of this proposal is to identify brain proteins predisposing to PTSD, AUD, or both as potential promising drug targets to support the development of novel treatments for these disorders.  

Our proposal builds on insights into the complex genetic architecture of PTSD and AUD gained through large genome-wide association studies (GWAS). GWAS tests allele frequency difference between cases and controls for individual single nucleotide polymorphisms (SNPs) to identify SNPs associated with disease. A genetic locus may have tens to dozens of SNPs associated with the disease. Disentangling which SNPs are important in predisposing to the disease versus those coincidentally associated because they are physically close to important genetic variant sites (i.e., due to linkage) is the next great challenge of human genetics. 

To address this challenge, we propose to leverage human brain proteomes to test the hypothesis that some genetic variants are associated with PTSD or AUD because they modulate brain protein expression in a way that predisposes to PTSD, AUD, or both. 

To test this hypothesis, we will integrate PTSD and AUD GWAS summary statistics, respectively, from participants of the Million Veteran Program and Psychiatric Genomics Consortium with human brain proteomes using the state-of-the-art analytical techniques to identify genes that modulate brain protein expression to predispose to PTSD, AUD, or both. We aim to identify genes with evidence consistent with being causal in PTSD or AUD, which means that they meet the following three conditions. First, the gene has one or more alleles strongly associated with the disease. Second, the brain protein expression regulated by proximal genetic variants (referred to as cis-regulated brain protein level) is associated with the disease. Third, the cis-regulated brain protein expression mediates the effect of the gene on the disease.  

The first aim of the proposal will focus on merging all available human brain proteomic and genetic data available from our own work and that of our collaborators to estimate effect of SNPs on human brain protein expression. In the second aim, we will integrate PTSD or AUD GWAS summary statistics with the brain protein expression data to identify potentially causal genes for PTSD, AUD, or both. Finally, in the third aim, we will test whether genetic regulation of protein expression for the causal genes identified in Aim 2 is similar in multiple specific brain regions relevant to PTSD or AUD (amygdala, hippocampus, anterior cingulate cortex, and nucleus accumbens).  

Findings from our proposed studies will identify specific genes and brain proteins as potential promising targets for further mechanistic study to support novel therapeutic development for PTSD, AUD, or both.  

 

## Acknowledgements

We are grateful to the many research volunteers who make our work possible, our
collaborators and financial support received from Veterans Administration, National
Institutes of Health, Emory University, The To Remember Foundation, American
Psychiatric Association, and the Brain and Behavior Foundation (formerly NARSAD).